regression-test/suites/nereids_rules_p0/mv/join_infer_derive/left_join_infer_and_derive.groovy

```diff
@@ -0,0 +1,239 @@
+        sql"""
+        CREATE MATERIALIZED VIEW ${mv_name} 
+        BUILD IMMEDIATE REFRESH AUTO ON MANUAL 
+        DISTRIBUTED BY RANDOM BUCKETS 2 
+        PROPERTIES ('replication_num' = '1') 
+        AS  
+        ${mv_sql}
+        """
+    }
+
+    def compare_res = { def stmt ->
+        sql "SET enable_materialized_view_rewrite=false"
+        def origin_res = sql stmt
+        logger.info("origin_res: " + origin_res)
+        sql "SET enable_materialized_view_rewrite=true"
+        def mv_origin_res = sql stmt
+        logger.info("mv_origin_res: " + mv_origin_res)
+        assertTrue((mv_origin_res == [] && origin_res == []) || (mv_origin_res.size() == origin_res.size()))
+        for (int row = 0; row < mv_origin_res.size(); row++) {
+            assertTrue(mv_origin_res[row].size() == origin_res[row].size())
+            for (int col = 0; col < mv_origin_res[row].size(); col++) {
+                assertTrue(mv_origin_res[row][col] == origin_res[row][col])
+            }
+        }
+    }
+
+    def mtmv_full_join_stmt = """
+        select l_shipdate, o_orderdate, l_partkey, l_suppkey, o_orderkey
+        from lineitem_left
+        full join orders_left
+        on lineitem_left.l_orderkey = orders_left.o_orderkey"""
+    def mv_name_1 = "mv_left"
+    def mtmv_stmt_list = [mtmv_full_join_stmt]
+
+    def mv_stmt_0 = """select t.l_shipdate, o_orderdate, t.l_partkey, t.l_suppkey, orders_left.o_orderkey 
+        from (select l_shipdate, l_partkey, l_suppkey, l_orderkey from lineitem_left where l_shipdate = '2023-10-17') t
+        left join orders_left 
+        on t.l_orderkey = orders_left.o_orderkey"""
+    def mv_stmt_1 = """select l_shipdate, t.o_orderdate, l_partkey, l_suppkey, t.o_orderkey
+        from lineitem_left  
+        left join (select o_orderdate,o_orderkey from orders_left where o_orderdate = '2023-10-17' ) t 
+        on lineitem_left.l_orderkey = t.o_orderkey"""
+    def mv_stmt_2 = """select l_shipdate, o_orderdate, l_partkey, l_suppkey, orders_left.o_orderkey 
+        from lineitem_left  
+        left join orders_left 
+        on lineitem_left.l_orderkey = orders_left.o_orderkey 
+        where l_shipdate = '2023-10-17'"""
+    def mv_stmt_3 = """select l_shipdate, o_orderdate, l_partkey, l_suppkey, orders_left.o_orderkey 
+        from lineitem_left  
+        left join orders_left 
+        on lineitem_left.l_orderkey = orders_left.o_orderkey 
+        where o_orderdate = '2023-10-17'"""
+    def mv_stmt_4 = """select l_shipdate, o_orderdate, l_partkey, l_suppkey, orders_left.o_orderkey 
+        from lineitem_left  
+        left join orders_left 
+        on lineitem_left.l_orderkey = orders_left.o_orderkey 
+        where o_orderkey = 1"""
+    def mv_stmt_5 = """select l_shipdate, o_orderdate, l_partkey, l_suppkey, orders_left.o_orderkey 
+        from lineitem_left  
+        left join orders_left 
+        on lineitem_left.l_orderkey = orders_left.o_orderkey 
+        where l_suppkey = 2"""
```

Comment:
Maybe we should have one more case like  
**mv_stmt_12**

as following  case  
select l_shipdate, o_orderdate, l_partkey, l_suppkey, orders_left.o_orderkey 
        from lineitem_left  
        left join orders_left  
        on lineitem_left.l_orderkey = orders_left.o_orderkey
        where l_suppkey = 2 and o_orderdate = '2023-10-17'

Reply:
new cases have been added